ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -66,31 +68,50 @@ public void generate(final Ds3ApiSpec spec, final FileUtils fileUtils, final Pat
     }
 
     private void generateCommands() throws IOException, TemplateException {
-        if (ConverterUtil.hasContent(spec.getRequests())) {
-            for (final Ds3Request request : spec.getRequests()) {
-                generateRequest(request);
-            }
-        }
-
         if (ConverterUtil.hasContent(spec.getTypes())) {
-            // Generate EnumConstant Types, which Element types can be dependant on
-            // TODO Determine if dependency matters at code generation time... probably not.
-            for (final Ds3Type typeEntry : spec.getTypes().values()) {
+            for (final Ds3Type ds3TypeEntry : spec.getTypes().values()) {
+                final Type typeEntry = TypeConverter.toType(ds3TypeEntry);
                 if (ConverterUtil.hasContent(typeEntry.getEnumConstants())) {
                     generateTypeEnumConstant(typeEntry);
                     generateTypeEnumConstantMatcher(typeEntry);
                 }
             }
 
+            // Generate TypeResponse parsers
+            final HashSet generatedTypesResponses = new HashSet();
+            int depth = 0;
+            while (generatedTypesResponses.size() != spec.getTypes().values().size() && depth++ < 10) {
```

Comment:
I assume that the depth < 10 is to prevent an infinite loop. As such, if depth does get that high, maybe an error or log message should be added to denote unexpected state.